src/ts/ConvivaAnalytics.ts

```diff
@@ -657,79 +698,127 @@ export class ConvivaAnalytics {
+
     if (!this.isSessionActive()) {
-      this.debugLog('skip custom event, no session existing', event);
+      this.debugLog('[ ConvivaAnalytics ] skip custom event, no session existing', event);
       return;
     }
 
     const eventAttributes = ObjectUtils.flatten(event);
     this.sendCustomPlaybackEvent(event.type, eventAttributes);
   };
 
-  private trackAdBreakStarted = (event: AdBreakEvent) => {
-    this.debugLog('[ ConvivaAnalytics ] adbreak started', event);
-    this.isAd = true;
+  // Fires before `onAdStarted`
+  private onAdBreakStarted = (event: AdBreakEvent) => {
+    this.debugLog('[ Player Event ] adbreak started', event);
 
-    const adPosition = AdBreakHelper.mapAdPosition(event.adBreak, this.player);
-
-    if (!this.isSessionActive()) {
-      // Don't report without a valid session (e.g., in case of a pre-roll, or post-roll ad)
-      return;
-    }
+    this.isAdBreak = true;
+    this.lastAdBreakEvent = event;
 
+    this.debugLog('[ ConvivaAnalytics ] report ad break started', event);
     this.convivaVideoAnalytics.reportAdBreakStarted(
       Conviva.Constants.AdType.CLIENT_SIDE,
       Conviva.Constants.AdPlayer.SEPARATE,
     );
   };
 
-  private onAdBreakFinished = (event: AdBreakEvent | ErrorEvent) => {
-    this.debugLog('[ ConvivaAnalytics ] adbreak finished', event);
-    this.isAd = false;
+  private onAdStarted = (event: AdEvent) => {
+    this.debugLog('[ Player Event ] ad started', event);
 
-    if (!this.isSessionActive()) {
-      // Don't report without a valid session (e.g., in case of a pre-roll, or post-roll ad)
-      return;
+    const adInfo = AdHelper.extractConvivaAdInfo(this.player, this.lastAdBreakEvent, event);
+    // It's already in kbps
+    const bitrateKbps = event.ad.data?.bitrate;
+
+    this.debugLog('[ ConvivaAdAnalytics ] report ad started', {
+      event,
+      adInfo,
+    });
+    this.convivaAdAnalytics.reportAdStarted(adInfo);
+
+    this.debugLog('[ ConvivaAdAnalytics ] report playing ad playback state');
+    this.convivaAdAnalytics.reportAdMetric(Conviva.Constants.Playback.PLAYER_STATE, Conviva.Constants.PlayerState.PLAYING);
+
+    if (bitrateKbps) {
+      this.debugLog('[ ConvivaAdAnalytics ] report ad bitrate', bitrateKbps);
+      this.convivaAdAnalytics.reportAdMetric(Conviva.Constants.Playback.BITRATE, bitrateKbps);
     }
+  }
+
+  // Fires before `onAdBreakFinished`
+  private onAdFinished = (event: AdEvent) => {
+    this.debugLog('[ Player Event ] ad finished', event);
+
+    this.debugLog('[ ConvivaAdAnalytics ] report ad ended', {
+      event,
+    });
+    this.convivaAdAnalytics.reportAdEnded();
+  }
 
+  // Fires before `onAdBreakFinished`
+  private onAdSkipped = (event: AdEvent) => {
+    this.debugLog('[ Player Event ] ad skipped', event);
+
+    this.debugLog('[ ConvivaAdAnalytics ] report ad skipped', event);
+    this.convivaAdAnalytics.reportAdSkipped();
+
+    this.onCustomEvent(event);
```

Comment:
Do we still need this custom event when we use `AdAnalytics` now? 🤔

Reply:
I would not remove anything if possible to not break something. 😅